pkg/client/controllers/servicemetadatawatcher_controller.go

```diff
@@ -320,17 +321,23 @@ func createServiceMetadataPatch(serviceId string, namespace string, field string
 // path is a list of keys separated by dots, e.g. "spec.template.spec.containers[0].image"
 // if the field is a slice, the last key must be in the form of "key[index]"
 func getNestedString(object interface{}, path []string) (string, bool, error) {
-	re := regexp.MustCompile(`^(.*)\[(\d+)]$`)
+	re := regexp.MustCompile(`^(.*)\[(\d+|[a-z]+)]$`)
 	var cpath []string
 	for i, key := range path {
 		m := re.FindStringSubmatch(key)
 		if len(m) > 0 {
 			cpath = append(cpath, m[1])
-			index, _ := strconv.Atoi(m[2])
 			slice, found, err := unstructured.NestedSlice(object.(map[string]interface{}), cpath...)
 			if !found || err != nil {
 				return "", false, err
 			}
+			index, err := strconv.Atoi(m[2])
+			if err != nil && m[2] != "last" {
```

Comment:
What does `last` value represent?

Reply:
Thanks for the review.
For our HTTPProxy and RequestProcessing, during the Egw->CGW migration, we would need to monitor their status. As I understood from my team and ethos team, there can be an array of statuses (and not only statuses, but other parameter arrays that we may need monitoring), so we would need to monitor only the last one of them. We have no way of knowing apriori the length of the status arrays, so we need the "last" entry in the array monitored - that's the significance of that string.

To give you an example of such array, that can be monitored in our HTTPProxy CR, we need the last authPolicy monitored:
routes:
  - authPolicy:
      context:
        policyId: default-api-key-validation
    conditions:
    - prefix: /api-key
    pathRewritePolicy:
      replacePrefix:
      - replacement: /get
    services:
    - name: httpbin
      port: 8080
  - authPolicy:
      context:
        policyId: custom-api-key-validation
    conditions:
    - prefix: /custom-api-key
    pathRewritePolicy:
      replacePrefix:
      - replacement: /get
    services:
    - name: httpbin
      port: 8080
  - authPolicy:
      context:
        policyId: open-api-key-validation
    conditions:
    - prefix: /open-api-key
    pathRewritePolicy:
      replacePrefix:
      - replacement: /get
    services:
    - name: httpbin
      port: 8080

Thanks a lot.